### PR TITLE
doc: revised document error

### LINKED ISSUE #33712: kernel/poll: no error happened when mutil-threads poll a same event at a same time.
**Describe the bug**
I took a look at Zephyr Specification https://docs.zephyrproject.org/latest/reference/kernel/other/polling.html#c.k_poll  about kernel/poll , I think there is a test point that was not tested before, the info as below:
image
It means if there are two thread polling a same event(sem/fifo) at a same time,  k_poll() immediately returns with the return value -EADDRINUSE. Is this right?
So I designed a testcase to test this, the code as below:

```
static K_SEM_DEFINE(test_sem, 0, 1);
static void poll_sem_t1(void *p1, void *p2, void *p3)
{
	printk("poll_sem_t1\n");

	int rc;
	struct k_poll_event *event = p1;

	rc = k_poll(event, 1, K_FOREVER);
	printk("rc1 = %d\n", rc);
	k_sem_take(&test_sem, K_FOREVER);
}

static void poll_sem_t2(void *p1, void *p2, void *p3)
{
	printk("poll_sem_t2\n");

	int rc;
	struct k_poll_event *event = p1;

	rc = k_poll(event, 1, K_FOREVER);
	printk("rc2 = %d\n", rc);
	k_sem_take(&test_sem, K_FOREVER);
}

void test_mutil_thread_poll(void)
{
	const int main_low_prio = 10;
	struct k_poll_event event;
	int old_prio = k_thread_priority_get(k_current_get());

	k_thread_priority_set(k_current_get(), main_low_prio);
	k_poll_event_init(&event, K_POLL_TYPE_SEM_AVAILABLE,
			  K_POLL_MODE_NOTIFY_ONLY, &test_sem);

	/* Spawn two threads */
	k_thread_create(&test_thread3, test_stack3,
			K_THREAD_STACK_SIZEOF(test_stack3),
			poll_sem_t1, &event, NULL, NULL, 10,
			K_INHERIT_PERMS, K_NO_WAIT);
	k_thread_create(&test_thread4, test_stack4,
			K_THREAD_STACK_SIZEOF(test_stack4),
			poll_sem_t2, &event, NULL, NULL, 10,
			K_INHERIT_PERMS, K_MSEC(10));


	k_sleep(K_MSEC(500));
	//printk("EADDRINUSE = %d\n", -EADDRINUSE);
	/* Make sem available */
	k_sem_give(&test_sem);
	k_sleep(K_MSEC(20));


	k_thread_priority_set(k_current_get(), old_prio);
}
```

and the output as below:
```
START - test_mutil_thread_poll
poll_sem_fifo_t1
poll_sem_fifo_t2
rc2 = 0
 PASS - test_mutil_thread_poll in 0.540 seconds
```

As you see, there is no error return value returned. So I think either the zephyr specification is wrong or there is something wrong with my code. Could you have any comments about this?

### PR DESCRIPTION
**Pull request by Zhaoningx**
Some description about poll are not right,
adding some new description to revise it.

Fixes #33712

Signed-off-by: Ningx Zhao <ningx.zhao@intel.com>

### CODE REVIEW

File: doc/reference/kernel/other/polling.rst [hunk 1]
```diff
@@ -123,11 +123,10 @@ amount of time, or the special values :c:macro:`K_NO_WAIT` and
...
 
-Only one thread can poll on a semaphore or a FIFO at a time. If a second thread
-tries to poll on the same semaphore or FIFO, :c:func:`k_poll` immediately
-returns with the return value -:c:macro:`EADDRINUSE`. In that case, if other
-conditions passed to :c:func:`k_poll` were met, their state will be set in
-the corresponding poll event.
+A list of pollers is offered on each semaphore or fifo and as many events
+can wait in it as the app wants.
+Warning: the waiters will be served in first-come-first-serve order and not
+prioritiy order.
```
Comment (enjiamai) [inline doc/reference/kernel/other/polling.rst:129]: typo, should be "a priority order".

**Review by enjiamai — CHANGES_REQUESTED** [review]

File: doc/reference/kernel/other/polling.rst [hunk 2]
```diff
@@ -123,11 +123,10 @@ amount of time, or the special values :c:macro:`K_NO_WAIT` and
 :c:macro:`K_FOREVER` to either not wait or wait until an event condition is
 satisfied and not sooner.
 
-Only one thread can poll on a semaphore or a FIFO at a time. If a second thread
-tries to poll on the same semaphore or FIFO, :c:func:`k_poll` immediately
-returns with the return value -:c:macro:`EADDRINUSE`. In that case, if other
-conditions passed to :c:func:`k_poll` were met, their state will be set in
-the corresponding poll event.
+A list of pollers is offered on each semaphore or fifo and as many events
```
Comment (nashif) [inline doc/reference/kernel/other/polling.rst:126]: FIFO

File: doc/reference/kernel/other/polling.rst [hunk 3]
```diff
@@ -123,11 +123,10 @@ amount of time, or the special values :c:macro:`K_NO_WAIT` and
...
+can wait in it as the app wants.
+Warning: the waiters will be served in first-come-first-serve order and not
```
[the remark this answers is not in the document]
Comment (Zhaoningx) [reply doc/reference/kernel/other/polling.rst]: Done, thanks.

File: doc/reference/kernel/other/polling.rst (hunk 1 above)
Comment (Zhaoningx) [reply doc/reference/kernel/other/polling.rst]: Done, thanks!

File: doc/reference/kernel/other/polling.rst (hunk 2 above)
Comment (Zhaoningx) [reply doc/reference/kernel/other/polling.rst]: Corrected it, thanks!

File: doc/reference/kernel/other/polling.rst (hunk 1 above)
Comment (enjiamai) [reply doc/reference/kernel/other/polling.rst]: please correct the typo of priority, thanks!

**Review by enjiamai — APPROVED** [review]

**Review by dcpleung — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**